src/UdpClient.hpp

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 Robotec.AI
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#ifdef _WIN32
+	#include <WinSock2.h>
+	#include <Ws2tcpip.h>
+	#pragma comment(lib, "ws2_32.lib")
+#else
+	#include <cerrno>
+	#include <sys/socket.h>
+	#include <arpa/inet.h>
+	#include <netinet/in.h>
+#endif // _WIN32
+
+#include "Logger.hpp"
+#include "RGLExceptions.hpp"
+
+#ifdef _WIN32
+	#define SOCKET_TYPE SOCKET
+#else
+	#define SOCKET_TYPE int
+	#define INVALID_SOCKET (-1)
+#endif // _WIN32
+
+struct UdpClient
+{
+	// Ctor cannot throw, because we want Dtor call if instance creation failed
```

Comment:
Not only can they throw, but it is also a good practice. [Learn more here](https://isocpp.org/wiki/faq/exceptions#ctors-can-throw)
Destructor won't be called, but you can solve this by wrapping the constructor into [function-try-block](https://en.cppreference.com/w/cpp/language/function-try-block). When an exception is caught, call a private `cleanup()` method (also called by the destructor) to release all acquired resources.